GANDLF/metrics/classification.py

```diff
@@ -25,6 +23,36 @@ def overall_stats(
         params["problem_type"] == "classification"
     ), "Only classification is supported for these stats"
 
+    def __convert_tensor_to_int(input_tensor: torch.Tensor) -> torch.Tensor:
+        """
+        Convert the input tensor to integer format.
+
+        Args:
+            input_tensor (torch.Tensor): The input tensor.
+
+        Returns:
+            torch.Tensor: The tensor converted to integer format.
+        """
+        return_tensor = input_tensor
+        if return_tensor.dtype != torch.long or return_tensor.dtype != torch.int:
+            return_tensor = return_tensor.long()
+        return return_tensor
+
+    # this is needed for a few metrics
+    # ensure that predictions and target are in integer format
+    prediction_wrap = __convert_tensor_to_int(prediction)
+    target_wrap = __convert_tensor_to_int(target)
+
+    # this is needed for auroc
+    # ensure that predictions are in integer format
+    prediction_wrap = prediction
```

Comment:
Same as above, I believe the predictions themselves are going to be modified

Reply:
Fixed.